### DISCUSSION TITLE
next build with static-export

### DISCUSSION THREAD

**Discussion by wind4gis** [vercel/next.js#63457] [Help] [answered]
### Summary

i face the same problem like this, i need next static export, and also support api router handler, how to fix it??
just use next build to compile static html, and also use the project to be backend?
the problem that using this same project to play backend, it will produce two same frontend, because next can not only serve api handler
https://github.com/vercel/next.js/issues/51990

**Comment by wind4gis** [comment]
i also re-write the backend code into server actions, without any api handler, only static-export and server actions,  it still no work...i can not run `next build` successfully, showing `Server Actions are not supported with static export.`

**Comment by icyJoseph** [comment, marked as answer]
Uhh this is a bit of a non-starter, because the static HTML export is meant to be used to create, well, a static site, to be served by another type of server, so called, static assets server, such as a CDN. 

That means that the next server is never constructed.

I think you need to rethink your design, to accommodate the Next.js server properly.

That being said, there's a dirty way to go about this, using custom extensions, I helped someone do something kind of similar before: 

- https://github.com/vercel/next.js/discussions/51891#discussioncomment-6297178

I'd say, that if you can't adapt the above to your use case, yourself, then you shouldn't probably do it, because it implies a high cost of maintenance and development time overhead, that you need to be fully aware when taking this approach.

**Reply by wind4gis** [reply]
yes, you're right...
now i set `{output: export}`  when building frontend static htmls, set `{output: standlone}` building backend api routes. it is not a good solution, yeah, it is a dirty way...
thx you...

**Reply by wind4gis** [reply]
@icyJoseph it seems no working in app router

**Reply by icyJoseph** [reply]
What is not working exactly? Is it the custom extension?

**Reply by icyJoseph** [reply]
in App Router Router Handlers are also statically optimized, so you might see stale data unless you do one of the opt-out techniques.

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching
